File: extract_pdf_emails.js
const fs = require('fs');
const pdf = require('pdf-parse');

async function extractEmails(filePath) {
    let dataBuffer = fs.readFileSync(filePath);
    let data = await pdf(dataBuffer);
    
    // Extract emails using regex
    const emailRegex = /[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}/g;
    const matches = data.text.match(emailRegex) || [];
    
    // Convert to lowercase and deduplicate
    return [...new Set(matches.map(e => e.toLowerCase()))];
}

async function run() {
    try {
        const caEmails = await extractEmails('c:/CET EXAM ONLINE/MSC CA Attendance.pdf');
        const csEmails = await extractEmails('c:/CET EXAM ONLINE/MSC CS Attendance.pdf');
        
        const allPdfEmails = [...new Set([...caEmails, ...csEmails])];
        
        console.log("Extracted total unique emails from PDFs:", allPdfEmails.length);
        
        fs.writeFileSync('c:/CET EXAM ONLINE/pdf_emails.json', JSON.stringify(allPdfEmails, null, 2));
        console.log("Saved to pdf_emails.json");
    } catch (e) {
        console.error("Error:", e);
    }
}

run();

File: extract_pdf_emails.py
import PyPDF2
import re
import json

def extract_emails(pdf_path):
    emails = []
    try:
        with open(pdf_path, 'rb') as file:
            reader = PyPDF2.PdfReader(file)
            for page in reader.pages:
                text = page.extract_text()
                if text:
                    found = re.findall(r'[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}', text)
                    emails.extend(found)
    except Exception as e:
        print(f"Error reading {pdf_path}: {e}")
    return [e.lower() for e in emails]

ca_emails = extract_emails('c:/CET EXAM ONLINE/MSC CA Attendance.pdf')
cs_emails = extract_emails('c:/CET EXAM ONLINE/MSC CS Attendance.pdf')

all_emails = list(set(ca_emails + cs_emails))
print(f"Total unique emails found: {len(all_emails)}")

with open('c:/CET EXAM ONLINE/pdf_emails.json', 'w') as f:
    json.dump(all_emails, f)
print("Saved to pdf_emails.json")
